app/assets/javascripts/lookup_keys.js

```diff
@@ -280,10 +280,13 @@ function build_match() {
 
 function toggle_lookupkey_hidden(checkbox) {
   var default_value = $('#'+checkbox.id.replace(/hidden_value$/, "default_value"));
-  var lookup_values = $(checkbox.closest('.fields')).find('.lookup_values>table [id$="value"]');
-  toggle_value_hidden(default_value, checkbox.checked);
-  lookup_values.each(function () {
-    toggle_value_hidden($(this, checkbox.checked));
+  var fields = $(checkbox.closest('.fields'));
+  var lookup_values = fields.find('.lookup_values>table [id$="value"]');
+  var hidden_lookup_values = fields.find('.lookup_values_fields_template').find('[id$="value"]');
+  shown = checkbox.checked;
```

Comment:
It is possible that this is not needed now that new values are already hidden/unhidden, try making the toggle work without passing the parameter.